openquake/commonlib/oqvalidation.py

```diff
@@ -170,6 +182,18 @@ def check_imts_gsims(self, gsims):
                         'The IMT %s is not accepted by the GSIM %s' %
                         (invalid_imts, gsim))
 
+            if 'site_model' not in self.inputs:
+                # look at the required sites parameters: they must have
+                # a valid value; the other parameters can keep a NaN
+                # value since they are not used by the calculator
+                for param in gsim.REQUIRES_SITES_PARAMETERS:
+                    param_name = self.siteparam[param]
+                    param_value = getattr(self, param_name)
+                    if (isinstance(param_value, float) and
+                            numpy.isnan(param_value)):
+                        raise ValueError('Please set a value for %s'
```

Comment:
Can you indicate in the error message the name of the GSIM that requires the parameter - it could be useful for the user to know which one.